collector/library/pod.js

```diff
@@ -156,46 +154,72 @@ class Pod {
         return this.accumulator[batch.id];
     }
 
-    /**
-     * As we are using an events emitter, this is the backbone of our collector application
-     * this method includes all the handlers for the various events we have in the system / per Pod of course.
-     */
-    registerHandlers() {
-        this.em.on(BATCH, async (batch) => {
-            console.log(`${chalk.greenBright('[NEW BATCH]')} (id: ${batch.id})`);
-            try {
-                const request = await fetch(`${this.targetUrl}/batch/${batch.id}?follow`);
-                const bytesAlreadySentToLogstashFromThisBatch = await this.getBytesSentFromThisParticularBatch({ batch });
-                let bytesSentAccumlator = 0;
-
-                request.body.on('data', async (data) => {
-                    let sendData = false;
-                    if (bytesAlreadySentToLogstashFromThisBatch > 0 && bytesSentAccumlator >= bytesAlreadySentToLogstashFromThisBatch) {
-                        sendData = true;
-                    }
+    async handleBatch(batch) {
+        console.log(`${chalk.greenBright('[NEW BATCH]')} (id: ${batch.id})`);
+        const request = await fetch(`${this.targetUrl}/batch/${batch.id}?follow`);
 
-                    if (bytesAlreadySentToLogstashFromThisBatch === 0) {
-                        sendData = true;
-                    }
+        const bytesAlreadySentToLogstashFromThisBatch = await this.getBytesSentFromThisParticularBatch({ batch });
+        let bytesSentAccumlator = 0;
 
-                    if (sendData) {
-                        await this.submitDataToLogstash({ data, batch });
-                    }
+        if (!request.ok) {
+            throw new Error('The request did not finish successfully!');
+        }
 
-                    bytesSentAccumlator += data.length;
-                });
+        await new Promise((res, rej) => {
+            this.em.on(DEAD_POD, () => {
+                rej(err);
+                request.body.destroy();
+            });
 
-            } catch (err) {
-                // The request got an error for some reason,
-                // The next round for new files scouting will pick up where we left off ;-)
-            }
-        });
+            request.body.on('data', async (data) => {
+                if (this.dead === true) { // Our pod has stopped, kill this request
```

Comment:
this is no longer needed. 

Notice, that if you want to guarantee that we never send any new bytes after a "pod" is killed, then you have a bug here because you just close the promise, destroy the request body, and continue to handle the incoming bytes ignoring the dead flag.